src/main/java/io/fabric8/maven/docker/assembly/DockerAssemblyManager.java

```diff
@@ -365,10 +374,49 @@ private Assembly extractAssembly(AssemblerConfigurationSource config) throws Moj
         }
     }
 
-
-
     // Archiver used to adapt for customizations
     private interface ArchiverCustomizer {
-        void customize(TarArchiver archiver) throws IOException;
+        TarArchiver customize(TarArchiver archiver) throws IOException;
+    }
+
+    private class TarPermissionsCustomizer implements ArchiverCustomizer {
+        ArchiverCustomizer innerCustomizer;
+
+        public TarPermissionsCustomizer(ArchiverCustomizer inner) {
+            innerCustomizer = inner;
+        }
+
+        @Override public TarArchiver customize(TarArchiver archiver) throws IOException {
+            if (innerCustomizer != null) {
+                archiver = innerCustomizer.customize(archiver);
+            }
+
+            // There is probably a better way to see if we're running on windows... but this should work.
+            if (File.separatorChar == '\\') {
+                TarArchiver newArchiver = new TarArchiver();
+                newArchiver.setDestFile(archiver.getDestFile());
+                newArchiver.setLongfile(TarLongFileMode.posix);
+
+                ResourceIterator resources = archiver.getResources();
+                while (resources.hasNext()) {
+                    ArchiveEntry ae = resources.next();
+                    String fileName = ae.getName();
+                    PlexusIoResource resource = ae.getResource();
+
+                    // See docker source:
+                    // https://github.com/docker/docker/blob/3d13fddd2bc4d679f0eaa68b0be877e5a816ad53/pkg/archive/archive_windows.go#L45
+                    int newMode = ae.getMode();
+                    newMode &= 0755;
+                    newMode |= 0111;
+
```

Comment:
Is it really that every file is marked as executable ? Why does this need to be ?